### ISSUE TITLE
std/text/unicode width does not return column-width 2 for emojis

### ISSUE THREAD

**Issue by erf** [koka-lang/koka#457]
I did expect the `width` function to return 2 for emojis when using the [EastAsianWidth.txt](https://www.unicode.org/Public/15.0.0/ucd/EastAsianWidth.txt) file.

```
  println(width("👾"))
```

this returns 1

Is this method supposed to work similar to the [display_width](
https://codeberg.org/dude_the_builder/ziglyph/src/commit/ac50ab06c91d2dd632ff4573c035dafe3b374aba/src/display_width.zig#L31) method of `ziglyph` or similar to this [Python wcwidth spesification](https://wcwidth.readthedocs.io/en/latest/specs.html) ? That is to give the rendered column width for modern terminal emulators using the latest Unicode standard?

**Comment by TimWhiting** [comment]
This is what the standard library documentation says:
```koka
// Return the column-width of a unicode character.
// Equivalent to ``wcwidth``
pub fun char/width( c : char ) : int {
  if (zero-widths.force.contains(c.int)) then 0
  elif (asian-wide.force.contains(c.int)) then 2
  else 1
}

// Return the total column-width of a string.
pub fun string/width( s : string ) : int {
  var total := 0
  s.foreach( fn(c) {
    total := total + c.width
  })
  total
}
```
So yes, I believe the intent is for terminal emulators as in the python wcwidth spec, however I'm not certain if it is currently up to date.

Also I would expect the following to print two utf16 characters, but it only does one utf32 character. I guess I'm less certain on the intended underlying representation for characters. I'll have to ask Daan.
 ``` "👾".slice.foreach(fn(c) c.println)```

**Comment by erf** [comment]
I'll just link this article here. It's a good read with some valuable links

https://mitchellh.com/writing/grapheme-clusters-in-terminals

**Comment by TimWhiting** [comment]
Thanks for the link!

**Comment by TimWhiting** [comment]
Great post. I'll have to look at the algorithm he references.

**Comment by erf** [comment]
Yeah i'm a beta tester on the Ghostty terminal (it's great!), and they have implemented [Mode 2027](https://github.com/contour-terminal/terminal-unicode-core) for proper Unicode handling
